hyperspy/_signals/eds_tem.py

```diff
@@ -353,37 +381,166 @@ def quantification(self,
         --------
         vacuum_mask
         """
-        if self.axes_manager.navigation_size == 0:
-            navigation_mask = None
-        elif isinstance(navigation_mask, float):
+        if isinstance(navigation_mask, float):
             navigation_mask = self.vacuum_mask(navigation_mask, closing).data
         elif navigation_mask is not None:
             navigation_mask = navigation_mask.data
 
+        xray_lines = [intensity.metadata.Sample.xray_lines[0] for intensity in intensities]
+        it = 0
+        pbar = tqdm(total=max_iterations+1)
+
         composition = utils.stack(intensities, lazy=False)
+        if take_off_angle == 'auto':
+            toa = self.get_take_off_angle()
+        else:
+            toa = take_off_angle
+
+        #determining illumination area for cross sections quantification.
+        if method == 'cross_section':
+            if probe_area == 'auto':
+                parameters = self.metadata.Acquisition_instrument.TEM
+                if probe_area in parameters:
+                    area = parameters.TEM.probe_area
+                else:
+                    if (self.axes_manager.navigation_dimension > 2 and
+                            navigation_axes is None):
+                        raise ValueError("With `probe_area='auto' and "
+                                         "navigation dimension > 2, you need "
+                                         "to specify the `navigation_axes` "
+                                         "parameter.")
+                    scales = []
+                    navigation_axes = self.axes_manager.navigation_axes
+                    for axis in navigation_axes:
+                        scales.append(
+                            axis.convert_to_units('nm', inplace=False)[0])
+                    if len(scales) == 1:
+                        probe_area = scales[0] * scales[0]
+                    elif len(scales) == 2:
+                        probe_area = scales[0] * scales[1]
+                    if scales[0] == 1 or scales[1] == 1:
+                        warnings.warn('Please note your probe_area is set to '
+                                      'the default value of 1 nm². The '
+                                      'function will still run. However if '
+                                      '1 nm² is not correct, please read the '
+                                      'user documentations for how to set '
+                                      'this properly.')
+
         if method == 'CL':
-            composition.data = utils_eds.quantification_cliff_lorimer(
-                composition.data, kfactors=factors,
-                mask=navigation_mask) * 100.
+            int_stack = utils.stack(intensities)
```

Comment:
It looks to me that this if/elif/else block could simplify by simply replace the quantification function and setting the argument accordingly:

```python
if method == 'CL':
    quantification_method = utils_eds.quantification_cliff_lorimer
    kwargs = {...}
elif method == 'zeta':
    quantification_method = utils_eds.quantification_zeta_factor
    kwargs = {...}

elif method == 'cross_section':
    quantification_method = utils_eds.quantification_cross_section
    kwargs = {...}
else:
    raise ValueError("Please specify method for quantification, "
                     "as 'CL', 'zeta' or 'cross_section'")
...
results = quantification_method(**kwargs)
```